File: Python/Sliding Window/Longest Subarray of 1's After Deleting One Element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        left = 0
        max_sub = 0
        zeros = 0
        for right in range(len(nums)):
            if nums[right] == 0:
                zeros += 1
            while zeros > 1:
                if nums[left] == 0:
                    zeros -= 1
                left += 1
            max_sub = max(max_sub, right - left)  # -1 bcz, we must delete one element
        return max_sub

File: Java/Sliding Window/Longest Subarray of 1's After Deleting One Element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int zeros=0;
        int max_sub=0;
        int left=0;
        for(int right=0;right<nums.length;right++){
            if(nums[right]==0){
                zeros++;
            }
            while(zeros>1){
                if(nums[left]==0)zeros--;
                left++;
            }
            max_sub=Math.max(max_sub,right-left); // -1 bcz, we must delete one element
        }
        return max_sub;
    }
}
